pkg/config/analysis/analyzers/gateway/secret.go

```diff
@@ -119,9 +121,28 @@ func isValidSecret(secret *resource.Instance) bool {
 	if err = xds.ValidateCertificate(certs.Cert); err != nil {
 		return false
 	}
+	if err = validateCertStrict(certs.Cert); err != nil {
+		return false
+	}
 	return true
 }
 
+func validateCertStrict(cert []byte) error {
+	cmd := exec.Command("openssl", "x509", "-noout", "-text")
```

Comment:
We cannot exec out in analyzer. Whatever we are trying to do here can be done in Go code?

Reply:
I haven't figured out an optimal way to do this in Go code, without using some 'magic' logic (which is not extensible). In some situations, `pem.Decode` may behaves differently than `openssl`, since `openssl` is stricter. However, for the above example I provided, `openssl` behaves the same as `pem.Decode` on a Linux system, while on a Mac, it throws a parse error. I actually preferred not to add this logic as well after finding the inconsistency.

Reply:
oy. that's a shame. so we are left with no real solution.